runtime/lua/vim/img/_backend/kitty.lua

```diff
@@ -0,0 +1,108 @@
+---@class vim.img.KittyBackend: vim.img.Backend
+local M = {}
+
+---For kitty, we need to write an image in chunks
+---
+---Graphics codes are in this form:
+---
+---    <ESC>_G<control data>;<payload><ESC>\
+---
+---To stream data for a PNG, we specify the format `f=100`.
+---
+---To simultaneously transmit and display an image, we use `a=T`.
+---
+---Chunking data (such as from over a network) requires the
+---specification of `m=0|1`, where all chunks must have a
+---value of `1` except the very last chunk.
+---@param data string
+local function write_seq(data)
+  local terminal = require('vim.img._terminal')
+
+  terminal.write(terminal.code.ESC .. '_G') -- Begin sequence
+  terminal.write(data) -- Primary data
+  terminal.write(terminal.code.ESC .. '\\') -- End sequence
+end
```

Comment:
Three separate writes means we are opening the TTY device, writing the data, and closing it three different times. We should be batching these writes as much as possible.